src/main/java/seedu/address/logic/commands/schedule/EditScheduleCommand.java

```diff
@@ -114,40 +133,43 @@ public boolean equals(Object other) {
         }
 
         // instanceof handles nulls
-        if (!(other instanceof RescheduleCommand)) {
+        if (!(other instanceof EditScheduleCommand)) {
             return false;
         }
 
         // state check
-        RescheduleCommand e = (RescheduleCommand) other;
-        return scheduleIndex.equals(e.scheduleIndex)
-                && editRescheduleDescriptor.equals(e.editRescheduleDescriptor);
+        EditScheduleCommand e = (EditScheduleCommand) other;
+        return clientIndex.equals(e.clientIndex)
+                && sessionIndex.equals(e.sessionIndex)
+                && editScheduleDescriptor.equals(e.editScheduleDescriptor);
     }
 
     /**
      * Stores the details to edit the Schedule with. Each non-empty field value will replace the
      * corresponding field value of the Schedule.
      */
-    public static class RescheduleDescriptor {
+    public static class EditScheduleDescriptor {
         private Index clientIndex;
         private Index sessionIndex;
+        private Index updateSessionIndex;
 
-        public RescheduleDescriptor() {}
+        public EditScheduleDescriptor() {}
 
         /**
          * Copy constructor.
          * A defensive copy of {@code tags} is used internally.
          */
-        public RescheduleDescriptor(RescheduleDescriptor toCopy) {
+        public EditScheduleDescriptor(EditScheduleDescriptor toCopy) {
             setClientIndex(toCopy.clientIndex);
             setSessionIndex(toCopy.sessionIndex);
+            setUpdatedSessionIndex(toCopy.updateSessionIndex);
         }
 
         /**
          * Returns true if at least one field is edited.
          */
         public boolean isAnyFieldEdited() {
-            return CollectionUtil.isAnyNonNull(clientIndex, sessionIndex);
+            return CollectionUtil.isAnyNonNull(clientIndex, sessionIndex, updateSessionIndex);
```

Comment:
should this takes in `updateSessionIndex` only? since the field that would be edited is session index (for now)

Reply:
yep sure I can remove that too, it was there for clarity sake when I was doing this